### PR TITLE
Feat: create stac items TDE-452

### PR DESCRIPTION
**Pull request by MDavidson17**
2
superceeds the already reviewed: https://github.com/linz/topo-imagery/pull/100

### CODE REVIEW

File: scripts/stac/imagery/item.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            raise Exception("incorrect initialising parameters must have 'stac' or 'id_ and path'")
+
+    def update_datetime(self, start_datetime: str, end_datetime: str) -> None:
+        self.stac["properties"] = {
+            "start_datetime": start_datetime,
+            "end_datetime": end_datetime,
+            "datetime": None,
+        }
+
+    def update_spatail(self, geometry: List[List[float]], bbox: List[float]) -> None:
```
Comment (paulfouquet) [inline scripts/stac/imagery/item.py:42]: typo here `spatail` -> `spatial`

File: scripts/stac/imagery/item.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        self.stac["geometry"] = {"type": "Polygon", "coordinates": [geometry]}
+        self.stac["bbox"] = bbox
+
+    def add_collection(self, title: str, path: str) -> None:
```
Comment (paulfouquet) [inline scripts/stac/imagery/item.py:46]: You could pass the `collection` object and manage getting the needed information from it in here

File: scripts/stac/tests/item_test.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+from scripts.files.files_helper import get_file_name_from_path
+from scripts.stac.imagery.item import ImageryItem
+
+
+def test_imagery_stac_item(mocker) -> None:  # type: ignore
```
Comment (paulfouquet) [inline scripts/stac/tests/item_test.py:5]: Should we add adding a collection to the item in this test?

**Review by paulfouquet — CHANGES_REQUESTED** [review]
a couple of comments

**Review by amfage — APPROVED** [review]

**Review by amfage — APPROVED** [review]

File: scripts/stac/imagery/collection.py [hunk 4]
```diff
@@ -4,22 +4,26 @@
 
 from scripts.stac.util.STAC_VERSION import STAC_VERSION
```
Comment (amfage) [inline scripts/stac/imagery/collection.py:6]:
```suggestion


```

**Review by amfage — APPROVED** [review]

**Review by amfage — APPROVED** [review]

**Pull request merged**